sourcegen-generator-bytecode/src/main/java/io/micronaut/sourcegen/ByteCodeGenerator.java

```diff
@@ -0,0 +1,1298 @@
+        var type = equalsStructurally.instance().type();
+        ExpressionDef equalsExpression;
+        if (type instanceof TypeDef.Array array) {
+            if (array.dimensions() > 1) {
+                equalsExpression = ClassTypeDef.of(Arrays.class)
+                    .invokeStatic(
+                        ARRAYS_DEEP_EQUALS,
+                        equalsStructurally.instance(),
+                        equalsStructurally.other()
+                    );
+            } else {
+                equalsExpression = ClassTypeDef.of(Arrays.class)
+                    .invokeStatic(
+                        "equals",
+                        TypeDef.Primitive.BOOLEAN,
+                        equalsStructurally.instance(),
+                        equalsStructurally.other()
+                    );
+            }
+        } else {
+            equalsExpression = equalsStructurally.instance().invoke(OBJECT_EQUALS, equalsStructurally.other());
+
+        }
+        pushExpression(generatorAdapter, context, equalsExpression, TypeDef.Primitive.BOOLEAN);
+    }
+
+    private void pushVariable(GeneratorAdapter generatorAdapter,
+                              Context context,
+                              VariableDef variableDef) {
+        if (variableDef instanceof VariableDef.Local localVariableDef) {
+            int index = context.locals.get(localVariableDef.name());
+            generatorAdapter.loadLocal(index);
+            return;
+        }
+        if (variableDef instanceof VariableDef.MethodParameter parameterVariableDef) {
+            if (context.methodDef == null) {
+                throw new IllegalStateException("Accessing method parameters is not available");
+            }
+            ParameterDef parameterDef = context.methodDef.getParameters().stream().filter(p -> p.getName().equals(parameterVariableDef.name())).findFirst().orElseThrow();
+            int parameterIndex = context.methodDef.getParameters().indexOf(parameterDef);
+            generatorAdapter.loadArg(parameterIndex);
+            return;
+        }
+        if (variableDef instanceof VariableDef.StaticField field) {
+            TypeDef owner = field.ownerType();
+            TypeDef fieldType = field.type();
+            generatorAdapter.getStatic(TypeUtils.getType(owner, context.objectDef), field.name(), TypeUtils.getType(fieldType, context.objectDef));
+            return;
+        }
+        if (variableDef instanceof VariableDef.Field field) {
+            if (context.objectDef == null) {
+                throw new IllegalStateException("Accessing 'this' is not available");
+            }
+            if (context.objectDef instanceof ClassDef classDef) {
+                if (!classDef.hasField(field.name()) && classDef.getProperties().stream().noneMatch(prop -> prop.getName().equals(field.name()))) {
+                    throw new IllegalStateException("Field " + field.name() + " is not available in [" + classDef + "]:" + classDef.getFields());
```

Comment:
exceptions thrown here should be processing exceptions unless these are handled somewhere, if so where?

Reply:
They are caught in the write method